O/OpenFOAM_com/build_tarballs.jl

```diff
@@ -0,0 +1,577 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+using Base.BinaryPlatforms
+const YGGDRASIL_DIR = "../.."
+include(joinpath(YGGDRASIL_DIR, "platforms", "mpi.jl"))
+
+name = "OpenFOAM_com"
+version = v"2312.0.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://develop.openfoam.com/Development/openfoam.git", "1d8f0d55f79e6488dae75e4b839e358a88af77b5")
+]
+
+# In order to set up OpenFOAM, we need to know the version of some of the
+# dependencies.
+const SCOTCH_VERSION = "6.1.3"
+const FFTW_VERSION = "3.3.10"
+
+# Bash recipe for building across all platforms
+script = raw"""
+
+cd ${WORKSPACE}/srcdir/openfoam
+git submodule update --init modules/cfmesh modules/avalanche
+
+# Adding -rpath-links #TODO need to automate for other platforms
+## For linux64
+LDFLAGS=""
+for dir in "" "/dummy" "/mpi-system"; do
+    LDFLAGS="${LDFLAGS} -Wl,-rpath-link=${PWD}/platforms/linux64GccDPInt32Opt/lib${dir}"
+done
+LDFLAGS="${LDFLAGS} -Wl,-rpath-link=${libdir}"
+
+# Set rpath-link in all C/C++ compilers
+sed -i "s|cc         := gcc\$(COMPILER_VERSION)|cc         := gcc\$(COMPILER_VERSION) ${LDFLAGS}|" wmake/rules/General/Gcc/c
+sed -i "s|CC         := g++\$(COMPILER_VERSION) -std=c++14|CC         := g++\$(COMPILER_VERSION) -std=c++14 ${LDFLAGS}|" wmake/rules/General/Gcc/c++
```

Comment:
It'd be better to use `cc`/`c++` as compilers, instead of forcing `gcc`/`g++`:
```suggestion
sed -i "s|cc         := gcc\$(COMPILER_VERSION)|cc         := cc\$(COMPILER_VERSION) ${LDFLAGS}|" wmake/rules/General/Gcc/c
sed -i "s|CC         := g++\$(COMPILER_VERSION) -std=c++14|CC         := c++\$(COMPILER_VERSION) -std=c++14 ${LDFLAGS}|" wmake/rules/General/Gcc/c++
```